saleor/graphql/product/filters.py

```diff
@@ -313,6 +318,7 @@ class ProductFilter(django_filters.FilterSet):
     stocks = ObjectTypeFilter(input_class=ProductStockFilterInput, method=filter_stocks)
     search = django_filters.CharFilter(method=filter_search)
     ids = GlobalIDMultipleChoiceFilter(field_name="id")
+    metadata = MetadataFilter()
```

Comment:
I would suggest creating the MetadataFilterBase class, inherited from django_filters.FilterSet. It should be placed somewhere in the `core` directory.  Then all filtres which should have metada field can use this class to automatically include it. Something similar we have for db models where we have `ModelWithMetadata` class to include meta fields. 
It should be similar to this example:
```
class Test(django_filters.FilterSet):
    metadata = MetadataFilter()
    class Meta:
        abstract=True

class ProductFilter(Test):
```

Reply:
At first I did that, but I didn't do it this way, probably because I'm still convinced that Metadata API is somewhat weird, to get my point can you have a look at the mentioned issue?